kamon-kafka-clients/src/main/scala/kamon/kafka/client/instrumentation/HasSpan.scala

```diff
@@ -0,0 +1,50 @@
+/*
+ * =========================================================================================
+ * Copyright © 2013-2019 the kamon project <http://kamon.io/>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License. You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ * =========================================================================================
+ */
+package kamon.kafka.client.instrumentation
+
+import kamon.trace.Span
+import org.apache.kafka.clients.consumer.ConsumerRecord
+
+import scala.util.Try
+
+trait HasSpan {
+  def maybeSpan: Option[Span]
+  def setSpan(span: Span)
+}
+
+object HasSpan {
+  class Mixin(@transient private var _span: Option[Span] = None) extends HasSpan {
+    override def maybeSpan: Option[Span] = _span
+    override def setSpan(span: Span): Unit = _span = Some(span)
+  }
+}
+
+object SpanExtractionSupport {
+
+  implicit class SpanOps[K, V](cr: ConsumerRecord[K, V]) {
```

Comment:
Few comments on this one:
- Is there a situation in which users themselves will want to write a Context into the ConsumerRecord? Maybe I'm missing something here.
- We need to add a method to get the Context, but we can also have one to get the Span, just because it is convenient. In that method please return just the `Span`, not `Option[Span]`.. if there is no span in the context Kamon will return a `Span.Empty` instead, same goes for the Context. We use these empty objects instead of returning Options of things because it provides a much more consistent usage across Java/Scala/Kotlin.
- I would recommend moving this to the `Client` companion object and breaking it down into two parts: methods on the companion object itself like `extractContext(consumerRecord)/extractSpan(consumerRecord)` and a `Syntax` implicit class that does what this class does. This is to allow Java users to access these functions but still keeping the cool stuff for the Scala folks :smile: